lib/src/main/java/com/nextcloud/android/sso/ui/UiExceptionManager.java

```diff
@@ -47,7 +50,21 @@ public static void showDialogForException(@NonNull Context context,
         final var optionalAction = exception.getPrimaryAction();
 
         if (optionalAction.isPresent()) {
-            showDialogForException(context, exception, actionText, (dialog, which) -> context.startActivity(optionalAction.get()));
+            showDialogForException(context, exception, actionText, (dialog, which) -> {
+                try {
```

Comment:
I am wondering whether we can check for an existing store *before* trying to launch it? But as I already said: It's an edge case so just reacting to the exception is fine for me, too.

Reply:
Probably "easier" and safer this way.. Otherwise we might get issues in case new stores pop up and we need to adjust the intents etc. And I believe we would need to register the "query" permission to even know about each store..

Reply:
It seems to be [possible](https://stackoverflow.com/questions/15407502/how-to-check-if-an-intent-can-be-handled-from-some-activity) without querying specific packages:

```java
intent.resolveActivity(getPackageManager()) != null;
```

So maybe we don't even need another exception type, but can just return a different `Intent` in `NextcloudFilesAppNotInstalledException`?

Reply:
@David-Development Please allow me to propose some changes for this PR: https://github.com/nextcloud/Android-SingleSignOn/pull/568

It will directly open the "correct" error dialog if no store is present.